Components/FlightControl/Inc/RocketSM.hpp

```diff
@@ -0,0 +1,295 @@
+    // For timing we want ~1/10th of a second or better.
+    RS_BURN,        // Main burn (vents closed MEV open) - 5-6 seconds (TBD) :
+                    
+
+    //-- COAST --
+    // Manual venting NOT ALLOWED
+    RS_COAST,       // Coasting (MEV closed, vents closed) - 30 seconds (TBD) ^ Vents closed applies here too, in part. Includes APOGEE
+
+    //-- DESCENT / POSTAPOGEE --
+    // Automatic Venting AND Vent Control ALLOWED
+    RS_DESCENT,  // Vents open (well into the descent)
+    RS_RECOVERY,  // Vents open, MEV closed, transmit all data over radio and accept vent commands
+
+    //-- RECOVERY / TECHNICAL --
+    RS_ABORT,       // Abort sequence, vents open, MEV closed, ignitors off
+    RS_NONE         // Invalid state, must be last
+};
+
+/**
+ * @brief Rocket State Commands, all fall under the GLOBAL_COMMAND -> STATE_ACTION umbrella
+ *
+ *        All commands in this list are only activated if the state allows them.
+ */
+enum RocketStateCommands
+{
+    //-- STATE TRANSITION --
+    RST_ABORT,      // Abort the rocket - transition to ABORT if available
+
+
+    RSC_MAX         // Invalid command, must be last
+};
+
+/**
+ * @brief External Rocket Control Commands, all fall under GLOBAL_COMMAND -> CONTROL_ACTION umbrella
+ *
+ *        State specific commands, must be all in-order to avoid duplicate command IDs
+ */
+enum RocketControlCommands
+{
+    //-- PRE-IGNITION and RECOVERY --
+    RSC_ABORT,       // Transition to ABORT state - available from all states except for IGNITION/LAUNCH/BURN
+    RSC_OPEN_VENT,   // Open the vent valve
+    RSC_CLOSE_VENT,  // Close the vent valve
+    RSC_OPEN_DRAIN,  // Open the drain valve
+    RSC_CLOSE_DRAIN, // Close the drain valve
+    RSC_MEV_CLOSE,   // Forces MEV to close - ONLY supported in states where it is safe to close the MEV
+
+    //-- PRELAUNCH --
+    RSC_BEGIN_FILL, // Transition to the FILL state
+
+    //-- FILL --
+    RSC_ARM_CONFIRM_1,   // Enable first ARM confirmation flag
+    RSC_ARM_CONFIRM_2,   // Enable second ARM confirmation flag
+    RSC_ARM_ACTION,      // Transition to the ARM state
+
+    //-- ARM/IGNITION/LAUNCH/BURN --
+    RSC_MANUAL_OVERRIDE_ENABLE, // For any states with locked out capability. If this flag is set using a command the NEXT command will be allowed to directly control valves/vents
+
+    //-- ARM --
+    RSC_POWER_TRANSITION_ONBOARD,      // Change power source to onboard
+    RSC_POWER_TRANSITION_EXTERNAL,     // Change power source to external power
+    RSC_READY_FOR_IGNITION, // Ready for ignition sequence - Transition to IGNITION state
+
+    //-- IGNITION --
+    RSC_CONFIRM_IGNITION,   // Confirm igniter actuation - Transition to LAUNCH state (MEV OPEN)
+    //* TBD - To ensure we don't get stuck unable to vent, we have override actions allowing a transition to ABORT
+    RSC_CRITICAL_IGNITION_ALLOW_ABORT, // Enable flag allowing us to abort in the ignition state
```

Comment:
again, unnecessary as per our last discussion

Reply:
We should be able to abort directly from ignition states? Doesn't that fundamentally go against the state diagram? 
![image](https://user-images.githubusercontent.com/78698227/208227054-3da0239a-898d-4a35-a3da-d78318393a15.png)